### PR TITLE
schedule: region fit replace does not make slice 

### LINKED ISSUE #5645: decrease the malloc for replace

### PR DESCRIPTION
**Pull request by bufferflies**
Signed-off-by: bufferflies <1045931706@qq.com>

### What problem does this PR solve?

Issue Number: Close #5645

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test

master:
image

 42 op/s          30047543 ns/op         4614343 B/op      54812 allocs/op

this pr:
image

50         23441817 ns/op         4296506 B/op      42006 allocs/op

Code changes

Side effects

Related changes

### Release note

```release-note
 None.
```

### CODE REVIEW

File: server/schedule/placement/fit.go [hunk 1]
```diff
@@ -66,11 +67,8 @@ func (f *RegionFit) Replace(srcStoreID uint64, dstStore *core.StoreInfo, region
 		return true
 	}
 
-	peers := newFitPeer(f.regionStores, region, fit.Peers, replaceFitPeerOpt(srcStoreID, dstStore))
```
Comment (lhy1024) [inline server/schedule/placement/fit.go:69]: why remove peers?

Comment (bufferflies) [reply server/schedule/placement/fit.go:69]: The `newFitPeer` will allocate memory to calculate the isolation score , it is too waste. So I use the `isolationStoreScore` avoid to allocate memory.

File: server/schedule/placement/fit.go [hunk 2]
```diff
@@ -398,6 +380,28 @@ func (p *fitPeer) matchRoleStrict(role PeerRoleType) bool {
...
 
+func isolationStoreScore(srcStoreID uint64, dstStore *core.StoreInfo, stores []*core.StoreInfo, labels []string) float64 {
+	var score float64
+	if len(labels) == 0 || len(stores) <= 1 {
+		return 0
+	}
+	for i := range stores {
+		store1 := stores[i]
+		if store1.GetID() == srcStoreID {
+			store1 = dstStore
```
Comment (lhy1024) [inline server/schedule/placement/fit.go:391]: why set it to dstStore

Comment (bufferflies) [reply server/schedule/placement/fit.go:391]: the current isolated score conside all the stores like [1,2,3], and the scheduler isolated score conside the new stores like [1,2,4], so the store-3 will be replace by store-4

File: server/schedule/placement/fit.go [hunk 3]
```diff
@@ -128,6 +126,8 @@ type RuleFit struct {
 	// IsolationScore indicates at which level of labeling these Peers are
 	// isolated. A larger value is better.
 	IsolationScore float64 `json:"isolation-score"`
+	// stores is the stores that the peers are placed in.s
```
Comment (rleungx) [inline server/schedule/placement/fit.go:129]:
```suggestion
	// stores is the stores that the peers are placed in.
```

File: server/schedule/placement/fit.go [hunk 4]
```diff
@@ -186,26 +186,13 @@ type fitWorker struct {
...
-		}
-	}
-}
-
-func newFitPeer(stores []*core.StoreInfo, region *core.RegionInfo, fitPeers []*metapb.Peer, opts ...fitPeerOpt) []*fitPeer {
+func newFitPeer(stores []*core.StoreInfo, region *core.RegionInfo, fitPeers []*metapb.Peer) []*fitPeer {
 	peers := make([]*fitPeer, len(fitPeers))
 	for i, p := range fitPeers {
-		peer := fitPeerPool.Get().(*fitPeer)
+		peer := &fitPeer{}
```
Comment (rleungx) [inline server/schedule/placement/fit.go:192]:
```suggestion
		peer := &fitPeer{
			Peer:     p,
			store:    getStoreByID(stores, p.GetStoreId()),
			isLeader: region.GetLeader().GetId() == p.GetId(),
		}
```

**Review by lhy1024 — APPROVED** [review]

File: server/schedule/placement/fit.go (hunk 4 above)
Comment (bufferflies) [reply server/schedule/placement/fit.go]: fixed

File: server/schedule/placement/fit.go (hunk 3 above)
Comment (bufferflies) [reply server/schedule/placement/fit.go]: fixed

**Review by rleungx — APPROVED** [review]

**Pull request merged**